ci2/ci/github.py

```diff
@@ -501,7 +540,8 @@ def config(self):
 
         seen_batch_ids = set()
         for pr in self.prs.values():
-            await pr._heal(batch_client, (merge_candidate is None or pr == merge_candidate), seen_batch_ids)
+            should_run = (merge_candidate is None or pr == merge_candidate or pr.should_build())
+            await pr._heal(batch_client, should_run, seen_batch_ids)
```

Comment:
I think we should change `run` to `is_merge_candidate` and push the `should_build` check should go into PR where is is more natural.